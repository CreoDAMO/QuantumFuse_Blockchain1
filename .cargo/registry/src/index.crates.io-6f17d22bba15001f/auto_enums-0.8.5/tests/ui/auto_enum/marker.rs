// SPDX-License-Identifier: Apache-2.0 OR MIT

use auto_enums::auto_enum;

#[auto_enum(Iterator, marker = foo)]
fn a(x: usize) -> impl Iterator<Item = i32> {
    #[auto_enum(Iterator, marker = foo)]
    //~ ERROR a custom marker name is specified that duplicated the name already used in the parent scope
    let _iter = match x {
        0 => 1..8,
        1 => return foo!(1..9),
        _ => (0..2).map(|x| x + 1),
    };

    match x {
        0 => 1..8,
        _ => (0..2).map(|x| x + 1),
    }
}

#[auto_enum(Iterator, marker = foo)]
fn b(x: usize) -> impl Iterator<Item = i32> {
    #[auto_enum(Iterator, marker = bar)]
    let _iter = match x {
        0 => 1..8,
        1 => return foo!(1..9),    // OK
        2 => return marker!(1..9), //~ ERROR cannot find macro `marker!` in this scope
        _ => (0..2).map(|x| x + 1),
    };

    match x {
        0 => 1..8,
        _ => (0..2).map(|x| x + 1),
    }
}

fn main() {}
