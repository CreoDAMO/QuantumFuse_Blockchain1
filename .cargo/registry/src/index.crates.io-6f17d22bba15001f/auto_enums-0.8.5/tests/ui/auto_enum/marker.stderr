error: a custom marker name is specified that duplicated the name already used in the parent scope
 --> tests/ui/auto_enum/marker.rs:7:36
  |
7 |     #[auto_enum(Iterator, marker = foo)]
  |                                    ^^^

error: cannot find macro `marker` in this scope
  --> tests/ui/auto_enum/marker.rs:27:21
   |
27 |         2 => return marker!(1..9), //~ ERROR cannot find macro `marker!` in this scope
   |                     ^^^^^^
   |
   = note: `marker` is in scope, but it is an attribute: `#[marker]`
